examples/systemcalls/systemcalls.c

```diff
@@ -39,15 +72,48 @@ bool do_exec(int count, ...)
     va_list args;
     va_start(args, count);
     char * command[count+1];
+    char * rest_of_args[count];
     int i;
     for(i=0; i<count; i++)
     {
         command[i] = va_arg(args, char *);
     }
+    
+    // checks if directory exists or not
+    DIR* dir = opendir(command[0]);
+    if(dir)
+    {
+    	closedir(dir);
+    }
+    else if(errno == ENOENT)
+    {
+    	return false;
+    }
+    
+    // add situation if number of arguments is 3
+    if(count == 3)
+    {
+    	DIR* dir1 = opendir(command[2]);
+    	if(dir1)
+    	{
+    		closedir(dir1);
+    	}
+    	else if(errno == ENOENT)
+    	{
+    		return false;
+    	}
+    }
```

Comment:
@abaa3023 I don't think you need the logic for directory checks above, I'm not sure I understand how this is working...

Reply:
It's probably some kind of workaround for the issues below.